### PR TITLE
Remove include type

### PR DESCRIPTION
**Pull request by Villetaneuse**
DO NOTE MERGE: Waiting to be rebased on Coq/Coq#19144 once/if it is merged.
Removing "Include Type" which was deprecated since 8.3.
"Include" now supports "Include type <+ type' with Definition foo := bar".
[Here](https://coq.gitlabpages.inria.fr/-/coq/-/jobs/4333244/artifacts/_build/default/doc/refman-html/language/core/modules.html?highlight=include#coq:cmd.Include) is the relevant part of the updated refman.

### CODE REVIEW

**Comment by SkySkimmer** [comment]
see also https://github.com/coq/coq/pull/11876

**Comment by Villetaneuse** [comment]
> see also #11876

The issue seems to still be there...
So what's the intended behaviour? Do we want the last command here to compile?
```coq
Module Type type1.
Parameter A : Prop.
End type1.

Module Type type2.
Parameter B : Prop.
End type2.

Module Type type3 := type1 <+ type2 with Definition B := True.
Print type3.

Module Type type3''.
Include type1 <+ type2 with Definition B := True.
(* Syntax error: '.' expected after [gallina_ext] (in [vernac_aux]). *)
```

**Comment by SkySkimmer** [comment]
@coqbot run full ci

**Comment by JasonGross** [comment]
> The issue seems to still be there...
> So what's the intended behaviour? Do we want the last command here to compile?

I think it would be nice to continue supporting "`Include` but instantiating some axioms", so we should just extend the syntax to support the final command.

**Comment by Villetaneuse** [comment]
@coqbot run full ci

**Comment by Villetaneuse** [comment]
I have made some progress on it. Waiting for feedback before triggering the full ci.

**Comment by Villetaneuse** [comment]
@coqbot run full ci

**Comment by Villetaneuse** [comment]
@coqbot run full ci

File: doc/changelog/08-vernac-commands-and-options/18482-remove-include-type.rst [hunk 1]
```diff
@@ -0,0 +1,9 @@
+- **Removed:**
+  the `Include Type` command after a long period of deprecation
+  (`#18482 <https://github.com/coq/coq/pull/18482>`_,
+  by Pierre Rousselin).
+- **Changed:**
+  The `Include` command can now include module types with a `with` declaration
```
Comment (jfehrle) [inline doc/changelog/08-vernac-commands-and-options/18482-remove-include-type.rst:6]:
```suggestion
  The :cmd:`Include` command can now include module types with a `with` clause (:n:`@with_declaration`)
```

Adds hyperlinks
`with` clause or declaration?

**Comment by Villetaneuse** [comment]
@coq/vernac-maintainers Do you think this PR is ready, or are there other things to take care of?

**Pull request closed without merging**